qiskit_dynamics/solvers/solver_classes.py

```diff
@@ -271,7 +445,9 @@ def solve(
         self,
         t_span: Array,
         y0: Union[Array, QuantumState, BaseOperator],
-        signals: Optional[Union[List[Signal], Tuple[List[Signal], List[Signal]]]] = None,
+        signals: Optional[
+            Union[List[Schedule], List[Signal], Tuple[List[Signal], List[Signal]]]
```

Comment:
Why `List[Schedule]`? Could we not have everything in one `Schedule`?

Reply:
For a list of simulations?

Reply:
Yeah it's for specifying a list of simulations. 

This is the main reason why the "specifying a list of simulations to `Solver.solve`" functionality was introduced. In all of the prior focused-on use cases for dynamics (optimization) it was natural to only specify a single simulation at a given time, but for pulse simulation there needs to be some layer at which looping takes place (as I believe it is common to submit a list of simulations?). We decided to put it here, ultimately because there is an opportunity to control JAX compilation when simulating pulses, so it's nice to have it accessible even at this layer of the interface.